lib/view_component/base.rb

```diff
@@ -516,12 +510,16 @@ def sidecar_files(extensions)
         (sidecar_files - [source_location] + sidecar_directory_files + nested_component_files).uniq
       end
 
+
+
       def cache_on(*args)
-        class_eval <<~RUBY, __FILE__, __LINE__ + 1
-          def __vc_cache_args
-            #{args}
-          end
-        RUBY
+        self.__vc_cache_dependencies.push(*args)
+      end
+
+      def view_cache_dependencies
+        return unless __vc_cache_dependencies.any?
```

Comment:
```suggestion
        return if __vc_cache_dependencies.none?
```

This has a lower cognitive load because it avoids having to flip the sentence in your read when you read any.

Reply:
I generally disagree with that in when using a single condition but when using multiple conditions I prefer an if statement